src/expr/proof_node_updater.cpp

```diff
@@ -82,19 +90,26 @@ void ProofNodeUpdater::processInternal(std::shared_ptr<ProofNode> pf,
   {
     cur = visit.back();
     visit.pop_back();
+    // don't continue visiting proof node
```

Comment:
I don't think this is the right place for this, since we haven't determined yet if cur is a new proof node to be processed.  Moreover I think the shouldContinue should be a flag on shouldUpdate (see other comment).